File: src/main/java/de/tectoast/emolga/buttons/GuildInviteButton.java
package de.tectoast.emolga.buttons;

import net.dv8tion.jda.api.events.interaction.component.ButtonInteractionEvent;

public class GuildInviteButton extends ButtonListener {

    public GuildInviteButton() {
        super("guildinvite");
    }

    @Override
    public void process(ButtonInteractionEvent e, String name) throws Exception {
        e.reply(e.getJDA().getGuildById(name).getDefaultChannel().createInvite().setMaxUses(1).complete().getUrl()).queue();
    }
}

File: src/main/kotlin/de/tectoast/emolga/buttons/GuildInviteButton.kt
package de.tectoast.emolga.buttons

import dev.minn.jda.ktx.coroutines.await
import net.dv8tion.jda.api.events.interaction.component.ButtonInteractionEvent

class GuildInviteButton : ButtonListener("guildinvite") {
    override suspend fun process(e: ButtonInteractionEvent, name: String) {
        e.jda.getGuildById(name)
            ?.let { g ->
                e.reply(g.defaultChannel!!.createInvite().setMaxUses(1).await().url).queue()
            } ?: e.reply("Invite failed, server not found").queue()
    }
}